python/llm/src/bigdl/llm/transformers/low_bit_linear.py

```diff
@@ -317,10 +348,14 @@ def forward(self, x: torch.Tensor):
             if x_2d.is_contiguous() is False:
                 x_2d = x_2d.contiguous()
             # current workaround to reduce first token latency of fp32 input
-            if x_2d.shape[0] > 1 and x_2d.dtype == torch.float32:
+            # sometimes fp16 cause nan and training instability
+            # disable the conversion when training
+            if self.conver_to_half and x_2d.shape[0] > 1 and x_2d.dtype == torch.float32:
                 x_2d = x_2d.half()
```

Comment:
is `half` fp16 only? we can try bf16 for training later

Reply:
Potentially user can still use fp16 or bf16 training by `autocast` even if `self.convert_to_half` is false. What we do here is to respect user's intent if he/she decides to use fp32.

Reply:
`autocast` often fails